src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class OperationStrategyImpl extends OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void execute(ShopStorage storage, FruitTransaction transaction) {
+        OperationHandler handler = operationHandlers.get(transaction.getOperation());
+        if (handler != null) {
+            handler.handle(storage, transaction);
+        } else {
+            System.out.println("Unknown operation: " + transaction.getOperation());
+        }
+    }
+}
```

Comment:
Refactor `OperationStrategy` to be an interface or abstract class, and ensure `OperationStrategyImpl` implements the method to retrieve the appropriate `OperationHandler` based on the transaction type.